packages/workbox-precaching/models/PrecachedDetailsModel.mjs

```diff
@@ -0,0 +1,76 @@
+import {_private} from 'workbox-core';
+
+const VERSION = 1;
+
+/**
+ * This model will track the relevant information of entries that
+ * are cached and their matching revision details.
+ */
+export default class PrecachedDetailsModel {
+  /**
+   * Construct a new model for a specific cache.
+   * @param {string} cacheName
+   */
+  constructor(cacheName) {
+    this._cacheName = _private.cacheNameProvider.getPrecacheName(cacheName);
+  }
+
+  /**
+   * Check if an entry is already cached. Returns false if
+   * the entry isn't cached or the revision has changed.
+   * @param {PrecacheEntry} precacheEntry
+   * @return {boolean}
+   */
+  async isEntryCached(precacheEntry) {
+    const revisionDetails = await this._getRevision(precacheEntry._entryId);
+    if (revisionDetails !== precacheEntry._revision) {
+      return false;
+    }
+
+    const openCache = await caches.open(this._cacheName);
+    const cachedResponse = await openCache.match(precacheEntry._request);
+    return !!cachedResponse;
+  }
+
+  /**
+   * Get the current revision details.
+   * @param {Object} entryId
+   * @return {Promise<string>}
```

Comment:
I think this should be `<string|null>` (or `<?string>`) since the promise can resolve to `null`.